app/src/main/java/de/hhn/aib/labsw/blackmirror/controller/API/Websockets/MirrorApiWebsockets.kt

```diff
@@ -30,12 +34,16 @@ class MirrorApiWebsockets : WebSocketListener(),MirrorApi {
             val jsonNode: JsonNode = mapper.readTree(text)
             val topic: String = jsonNode.get("topic").textValue()
             requireNotNull(jsonNode.get("payload")) { "wrong json format" }
-            val listenersList: ArrayList<ApiListener>? = listeners[topic]
-            listenersList?.forEach { element : ApiListener ->
-                try {
-                    element.dataReceived(topic, jsonNode)
-                } catch (e: NullPointerException) {
-                    listenersList.remove(element)
+            val listenersList = listeners[topic]
+            if(listenersList != null) {
+                val listIterator = listenersList.iterator()
+                while (listIterator.hasNext()) {
+                    val element = listIterator.next()
+                    try {
+                        element.dataReceived(topic, jsonNode)
+                    } catch (e: NullPointerException) {
```

Comment:
`ApiListener::dataReceived()` is not annotated with `@throws NullPointerException` so why do we catch it? Because of `element`? `element` is not nullable either!